cpp/ps0005/ps0005_soin_cpp.cpp

```diff
@@ -1,31 +1,43 @@
 #include <iostream>
-#include<conio.h>
+#include <conio.h>
 using namespace std;
 int main(){
-	int number1, number2;
-	char x;
+	// prompting user for 2 operands and an operator 1.5/2
+	double number1, number2;
+	char op;
 	cout << "enter the two operands\n";
 	cin >> number1;
 	cin >> number2;
 	cout << "enter the operator + - % * / \n";
-	cin >> x;
-
-	if (x == '+')
+	cin >> op;
 
+	// checking user inputs and processing 2/4
+	// printing the result if all data is valid 1/1
+	if (op == '+')
 		cout << number1<<"+"<< number2<<"="<<number1+number2;
-	else if (x == '-')
+	else if (op == '-')
 		cout << number1 << "-" << number2 << "=" << number1 - number2;
-	else if (x == '%')
-
-		cout << number1 << "%" << number2 << "=" << number1 % number2;
-
-	else if (x == '*')
-
+	else if (op == '*')
 		cout << number1 << "*" << number2 << "=" << number1 * number2;
-	else if (x == '/')
-
-		cout << number1 << "/" << number2 << "=" << number1 / number2;
+	else if (op == '/') {
+		if (number1 == 0 && number2 == 0)
+			cout << "Infinite number of solutions: (0 / 0) is Undefined Quantity.\n";
+		else if (number1 != 0 && number2 == 0)
+			cout << "Math error : cannot divide by 0 \n";
+		else
+			cout << number1 << "/" << number2 << "=" << number1 / number2;
+	} // 0/1
+	else if (op == '%')
+	{
+		if (number1 != (int)number1 || number2 != (int)number2)
+			cout << "Cannot execute modulo operation for floating point numbers.\n";
+		else
+			cout << number1 << "%" << number2 << "=" << (int)number1 % (int)number2;
+	} // 0/1
 	else
-		cout << "Invalid character for operator \n";
+		cout << "Invalid character for operator \n"; // 1/1
+	// 1/1
+	
 	_getch();
-}
\ No newline at end of file
+}
+// good work 👍 5.5/10
```

Comment:
- [x] program is in the correct path
- [x] no syntax errors while compilation
- [ ] formatted code to be clear and readable
- [ ] no bugs or warnings while execution
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case

you can check the [problem solution](https://onlinegdb.com/2mF7i4R8H) for more